File: backend/src/main/java/com/team05/petmeeting/domain/ads/dto/GeminiRequest.java
package com.team05.petmeeting.domain.ads.dto;

import java.util.List;

public record GeminiRequest(List<Content> contents) {
    public record Content(List<Part> parts) {}
    public record Part(String text) {}

    public static GeminiRequest from(String prompt) {
        return new GeminiRequest(List.of(new Content(List.of(new Part(prompt)))));
    }
}

File: backend/src/main/java/com/team05/petmeeting/domain/ads/dto/GeminiRequest.kt
package com.team05.petmeeting.domain.ads.dto

data class GeminiRequest(
    @JvmField val contents: List<Content>
) {
    fun contents(): List<Content> = contents

    data class Content(
        @JvmField val parts: List<Part>
    ) {
        fun parts(): List<Part> = parts
    }

    data class Part(
        @JvmField val text: String?
    ) {
        fun text(): String? = text
    }

    companion object {
        @JvmStatic
        fun from(prompt: String?): GeminiRequest {
            return GeminiRequest(listOf(Content(listOf(Part(prompt)))))
        }
    }
}
